internal/command/secrets/deploy.go

```diff
@@ -0,0 +1,59 @@
+package secrets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/superfly/flyctl/client"
+	"github.com/superfly/flyctl/flaps"
+	"github.com/superfly/flyctl/internal/appconfig"
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/flag"
+)
+
+func newDeploy() (cmd *cobra.Command) {
+	const (
+		long  = `Deploy staged secrets for an application`
+		short = long
+		usage = "deploy [flags]"
+	)
+
+	cmd = command.New(usage, short, long, runDeploy, command.RequireSession, command.RequireAppName)
+
+	flag.Add(cmd,
+		sharedFlags,
+	)
+
+	return cmd
+}
+
+func runDeploy(ctx context.Context) (err error) {
+	client := client.FromContext(ctx).API()
+	appName := appconfig.NameFromContext(ctx)
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		return err
+	}
+
+	flapsClient, err := flaps.New(ctx, app)
+	if err != nil {
+		return fmt.Errorf("could not create flaps client: %w", err)
+	}
+
+	machines, _, err := flapsClient.ListFlyAppsMachines(ctx)
+	if err != nil {
+		return err
+	}
+
+	if !app.Deployed && len(machines) == 0 {
```

Comment:
do we need this check here? In fact, if we do it should be after the `app.PlatformVersion == "machines"` check

Reply:
> Should this maybe be an or?

So that logic taken from DeploySecrets, but the error messaging there is not helpful there.

```
# DeploySecrets
	if !app.Deployed && len(machines) == 0 {
		fmt.Fprintln(out, "Secrets are staged for the first deployment")
		return nil
	}
``` 

>  it should be after the app.PlatformVersion == "machines" check

Yeah, that would cover our bases more and make the && make more sense.